src/mcp/tools/core/list_apps.ts

```diff
@@ -23,10 +23,15 @@ export const list_apps = tool(
     },
   },
   async ({ platform }, { projectId }) => {
-    const apps = await listFirebaseApps(
-      projectId!,
-      (platform?.toUpperCase() as AppPlatform) ?? AppPlatform.ANY,
-    );
-    return toContent(apps);
+    try {
+      const apps = await listFirebaseApps(
+        projectId!,
+        platform === "all" ? AppPlatform.ANY : (platform.toUpperCase() as AppPlatform),
+      );
+      return toContent(apps);
+    } catch (err: any) {
+      const originalMessage = err.original ? `: ${err.original.message}` : "";
```

Comment:
Wonder if this logic is worth porting to the upstream exception handler.

It feels like a generic enough improvement.

https://github.com/firebase/firebase-tools/blob/master/src/mcp/index.ts#L223